src/JustSaying/Fluent/AccountAddressProvider.cs

```diff
@@ -0,0 +1,100 @@
+        /// <param name="topicNamingConvention">A <see cref="ITopicNamingConvention"/> to use for producing a topic name from a message type</param>
+        public AccountAddressProvider(string accountId, string regionName, IQueueNamingConvention queueNamingConvention, ITopicNamingConvention topicNamingConvention)
+        {
+            _accountId = accountId;
+            _regionEndpoint = RegionEndpoint.GetBySystemName(regionName);
+            _queueNamingConvention = queueNamingConvention;
+            _topicNamingConvention = topicNamingConvention;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="TopicAddress"/> within the current account from <see cref="T"/> by using the topic naming convention.
+        /// </summary>
+        /// <typeparam name="T">Message type</typeparam>
+        /// <returns>The <see cref="TopicAddress"/> for this message type.</returns>
+        public TopicAddress GetTopicAddressByConvention<T>()
+        {
+            return GetTopicAddress(_topicNamingConvention.TopicName<T>());
+        }
+
+        /// <summary>
+        /// Creates a <see cref="TopicAddress"/> in the current account with name specified with <see cref="topicName"/>.
+        /// </summary>
+        /// <param name="topicName">The topic name.</param>
+        /// <returns>The <see cref="TopicAddress"/> for this topic.</returns>
+        public TopicAddress GetTopicAddress(string topicName)
+        {
+            return new TopicAddress
+            {
+                TopicArn = new Arn
+                {
+                    Partition = _regionEndpoint.PartitionName,
+                    Service = "sns",
+                    Region = _regionEndpoint.SystemName,
+                    AccountId = _accountId,
+                    Resource = topicName
+                }.ToString()
+            };
+        }
+
+        /// <summary>
+        /// Creates a <see cref="QueueAddress"/> within the current account from <see cref="T"/> by using the queue naming convention.
+        /// </summary>
+        /// <typeparam name="T">Message type</typeparam>
+        /// <returns>The <see cref="QueueAddress"/> for this message type.</returns>
+        public QueueAddress GetQueueAddressByConvention<T>()
+        {
+            return GetQueueAddress(_queueNamingConvention.QueueName<T>());
+        }
+
+        /// <summary>
+        /// Creates a <see cref="QueueAddress"/> in the current account with name specified with <see cref="queueName"/>.
+        /// </summary>
+        /// <param name="queueName">The queue name.</param>
+        /// <returns>The <see cref="QueueAddress"/> for this queue.</returns>
+        public QueueAddress GetQueueAddress(string queueName)
+        {
+            return new QueueAddress
+            {
+                QueueUrl = new Uri($"https://sqs.{_regionEndpoint.SystemName}.{_regionEndpoint.PartitionDnsSuffix}/{_accountId}/{queueName}"),
```

Comment:
Might be nicer to use `UriBuilder` to do this, then you'd only need to concat the host.

Reply:
This made me chuckle:
https://github.com/aws/aws-sdk-net/blob/475822dec5e87954b7a47ac65995714ae1f1b115/sdk/src/Services/S3Control/Custom/Internal/S3ArnUtils/S3OutpostResource.cs#L148-L149